### PR TITLE
feat(framework) Add template to create a Flower Baseline

### PR DESCRIPTION
**Pull request by jafermarq**
Adds template creation logic for a new Flower Baseline.

Screenshot 2024-08-08 at 16 51 00

It reutilises components from the `PoTorch` template. It will generate the following directory structure:

```
.
├── LICENSE
├── README.md
├── mycoolbaseline # name given during `flwr new`
│   ├── __init__.py
│   ├── client_app.py
│   ├── dataset.py
│   ├── model.py
│   ├── models.py
│   ├── strategy.py
│   └── utils.py
└── pyproject.toml
```

### CODE REVIEW

File: src/py/flwr/cli/new/new.py [hunk 1]
```diff
@@ -259,6 +262,21 @@ def new(
                 "template": f"app/code/task.{framework_str}.py.tpl"
             }
 
+        if is_baseline_project:
+            # Indluce additional files for baseline template
```
Comment (tanertopal) [inline src/py/flwr/cli/new/new.py:266]:
```suggestion
            # Include additional files for baseline template
```

**Review by tanertopal — COMMENTED** [review]
Lgtm! I only made one small suggestion for a typo.

File: src/py/flwr/cli/new/templates/app/README.baseline.md.tpl [hunk 2]
```diff
@@ -0,0 +1,128 @@
+---
+title: title of the paper
+url: https://arxiv.org/abs/2007.14390 # update with the link to your paper
+labels: [label1, label2] # please add between 4 and 10 single-word (maybe two-words) labels (e.g. system heterogeneity, image classification, asynchronous, weight sharing, cross-silo). Do not use "". Remove this comment once you are done.
+dataset: [dataset1, dataset2] # list of datasets you include in your baseline. Do not use "". Remove this comment once you are done.
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/README.baseline.md.tpl:5]: Should we add a `TODO` at the beginning of each of those comments? We can then easily search if any TODOs are left

File: src/py/flwr/cli/new/templates/app/code/client.baseline.py.tpl [hunk 3]
```diff
@@ -0,0 +1,58 @@
+"""$project_name: A Flower $framework_str app."""
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/code/client.baseline.py.tpl:1]: In other templates, we use `/` (same comment for all the other files that are missing the `/`):
```suggestion
"""$project_name: A Flower / $framework_str app."""
```

File: src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl [hunk 4]
```diff
@@ -0,0 +1,142 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "$package_name"
+version = "1.0.0"
+description = ""
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl:8]: Do we expect users to write something here? Should we add a TODO?

File: src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    "types-requests==2.31.0.20240125",
+]
+
+[tool.isort]
+line_length = 88
+indent = "    "
+multi_line_output = 3
+include_trailing_comma = true
+force_grid_wrap = 0
+use_parentheses = true
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl:40]: Isn't there a newer version of this where isort supports a `black` profile?

File: src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+fraction-fit = 0.5
+local-epochs = 1
+
+[tool.flwr.federations]
+default = "local-simulation"
+
+[tool.flwr.federations.local-simulation]
+options.num-supernodes = 10
+options.backend.client-resources.num-cpus = 2
+options.backend.client-resources.num-gpus = 0.0
```
[the remark this answers is not in the document]
Comment (jafermarq) [reply src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl:142]: I thought of keeping it as in the rest of templates and (migrated) examples

File: src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl (hunk 4 above)
Comment (jafermarq) [reply src/py/flwr/cli/new/templates/app/pyproject.baseline.toml.tpl:8]: yes, just mentioned it in the `README.md` that gets generated.

File: src/py/flwr/cli/new/templates/app/code/dataset.baseline.py.tpl [hunk 7]
```diff
@@ -0,0 +1,36 @@
+"""$project_name: A Flower $framework_str app."""
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/code/dataset.baseline.py.tpl:1]:
```suggestion
"""$project_name: A Flower / $framework_str app."""
```

File: src/py/flwr/cli/new/templates/app/code/model.baseline.py.tpl [hunk 8]
```diff
@@ -0,0 +1,80 @@
+"""$project_name: A Flower $framework_str app."""
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/code/model.baseline.py.tpl:1]:
```suggestion
"""$project_name: A Flower / $framework_str app."""
```

File: src/py/flwr/cli/new/templates/app/code/server.baseline.py.tpl [hunk 9]
```diff
@@ -0,0 +1,46 @@
+"""$project_name: A Flower $framework_str app."""
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/code/server.baseline.py.tpl:1]:
```suggestion
"""$project_name: A Flower / $framework_str app."""
```

File: src/py/flwr/cli/new/templates/app/code/strategy.baseline.py.tpl [hunk 10]
```diff
@@ -0,0 +1 @@
+"""$project_name: A Flower $framework_str app."""
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/code/strategy.baseline.py.tpl:1]:
```suggestion
"""$project_name: A Flower / $framework_str app."""
```

File: src/py/flwr/cli/new/templates/app/code/utils.baseline.py.tpl [hunk 11]
```diff
@@ -0,0 +1 @@
+"""$project_name: A Flower $framework_str app."""
```
Comment (danieljanes) [inline src/py/flwr/cli/new/templates/app/code/utils.baseline.py.tpl:1]:
```suggestion
"""$project_name: A Flower / $framework_str app."""
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**